experiments/AtmosLES/bomex.jl

```diff
@@ -514,19 +510,33 @@ function main()
         nothing
     end
 
-    norm_ρ_Q₀ = norm(solver_config.Q[:, 1, :])
-    cb_check_mass_cons =
-        GenericCallbacks.EveryXSimulationSteps(500) do (init = false)
-            norm_ρ = norm(solver_config.Q[:, 1, :])
-            Δρ_ratio = (norm_ρ - norm_ρ_Q₀) / norm_ρ_Q₀
-            @test isapprox(Δρ_ratio, FT(0); atol = 1e-3)
+    # State variable
+    Q = solver_config.Q
+    # Volume geometry information
+    vgeo = driver_config.grid.vgeo
+    M = vgeo[:,Grids._M,:]
+    # Unpack prognostic vars
+    ρ₀ = Q.ρ
+    ρe₀ = Q.ρe
```

Comment:
This won't work here -- `Q` is an `MPIStateArray` (like `vgeo` above). Are you looking for all `ρ` for the entire grid?

Reply:
Yes